kratos/testing/tester.cpp

```diff
@@ -228,8 +228,7 @@ namespace Kratos
 		{
 			// creating the regex pattern replacing * with ".*"
 
-#if defined(__GNUC__) &&( __GNUC__ < 4 || \
-              (__GNUC__ == 4 && (__GNUC_MINOR__ < 9))) 
+#if __cplusplus <= 199711L
```

Comment:
Exactly as @RiccardoRossi saying, gcc 4.8 which is one of the popular old one does not support it and for me this function wasn't enough for drop it